packages/rich-text/src/plugins/CommandPalette/components/CommandList.tsx

```diff
@@ -0,0 +1,200 @@
+import * as React from 'react';
+
+import { Popover, Stack, SectionHeading, ScreenReaderOnly, Flex } from '@contentful/f36-components';
+import { PlateEditor } from '@udecode/plate-core';
+import { cx } from 'emotion';
+import isHotkey from 'is-hotkey';
+
+import { useSdkContext } from '../../../SdkProvider';
+import { useCommands } from '../useCommands';
+import styles from './CommandList.styles';
+
+export interface CommandListProps {
+  query: string;
+  editor: PlateEditor;
+}
+
+export const useCommandList = (commandItems, container) => {
+  const [selectedItem, setSelectedItem] = React.useState<string>(() => {
+    // select the first item on initial render
+    if ('group' in commandItems[0]) {
+      return commandItems[0].commands[0].id;
+    }
+    return commandItems[0].id;
+  });
+
+  React.useEffect(() => {
+    if (!container || !container.current) {
+      return;
+    }
+    const buttons = Array.from(container.current.querySelectorAll('button')) as HTMLButtonElement[];
+    const currBtn = buttons.find((btn) => btn.id === selectedItem);
+    const currIndex = currBtn ? buttons.indexOf(currBtn) : 0;
+
+    if (!currBtn && buttons.length) {
```

Comment:
NIT: I'd suggest extracting this into a bool variable and giving it a self-explanatory name to make it easy to understand what the condition represents at a glance: `!currBtn && buttons.length`